### PR TITLE
client: both Support names and config path

### LINKED ISSUE #5885: client: both Support names and config path
## Enhancement Task
The global config needs to support both names and configuration paths, not only for compatibility but also to make more sense.

### PR DESCRIPTION
**Pull request by HuSharp**
Signed-off-by: husharp <jinhao.hu@pingcap.com>

### What problem does this PR solve?

Issue Number: Close #5885, Ref #https://github.com/pingcap/tiflow/issues/8110, should merged after #5887 #5888 and https://github.com/pingcap/kvproto/pull/1051

### What is changed and how does it work?

```commit-message
The global config needs to support both names and configuration paths, not only for compatibility but also to make more sense.
```

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: server/grpc_service.go [hunk 1]
```diff
@@ -1907,13 +1919,34 @@ func (s *GrpcServer) StoreGlobalConfig(_ context.Context, request *pdpb.StoreGlo
 // LoadGlobalConfig load global config from etcd
 func (s *GrpcServer) LoadGlobalConfig(ctx context.Context, request *pdpb.LoadGlobalConfigRequest) (*pdpb.LoadGlobalConfigResponse, error) {
 	configPath := request.GetConfigPath()
+	if configPath == "" {
+		configPath = globalConfigPath
+	}
+	// `TiKV` cannot use `Names` field Since item value needs to support marshal of different struct types,
+	// it should be set to bytes instead of string.
+	// But for CDC compatibility, we need to keep the Value field.
+	if request.Names != nil {
```
Comment (JmPotato) [inline server/grpc_service.go:1928]: What if the `Names` is not nil but empty?

```suggestion
	if len(request.GetNames()) != 0 {
```

Comment (HuSharp) [reply server/grpc_service.go:1928]: I don't know what case will names be []{""}? In this case, why not just pass nil🤔

File: server/grpc_service.go [hunk 2]
```diff
@@ -1881,16 +1882,24 @@ func checkStream(streamCtx context.Context, cancel context.CancelFunc, done chan
 	<-done
 }
 
+const globalConfigPath = "/global/config/"
+
 // StoreGlobalConfig store global config into etcd by transaction
 func (s *GrpcServer) StoreGlobalConfig(_ context.Context, request *pdpb.StoreGlobalConfigRequest) (*pdpb.StoreGlobalConfigResponse, error) {
+	configPath := request.GetConfigPath()
+	if configPath == "" {
```
[the remark this answers is not in the document]
Comment (HuSharp) [reply server/grpc_service.go:1891]: add it at `globalConfigPath`

**Review by glorv — COMMENTED** [review]
LGTM

**Comment by HuSharp** [comment]
@JmPotato @nolouch @CabinfeverB friendly ping, thx! :)

File: server/grpc_service.go [hunk 3]
```diff
@@ -1947,7 +1988,7 @@ func (s *GrpcServer) WatchGlobalConfig(req *pdpb.WatchGlobalConfigRequest, serve
 
 			cfgs := make([]*pdpb.GlobalConfigItem, 0, len(res.Events))
 			for _, e := range res.Events {
-				cfgs = append(cfgs, &pdpb.GlobalConfigItem{Name: string(e.Kv.Key), Value: string(e.Kv.Value), Kind: pdpb.EventType(e.Type)})
+				cfgs = append(cfgs, &pdpb.GlobalConfigItem{Name: string(e.Kv.Key), Payload: e.Kv.Value, Kind: pdpb.EventType(e.Type)})
```
Comment (CabinfeverB) [inline server/grpc_service.go:1991]: Add a comment to show that Watch only returns Payload

**Review by CabinfeverB — APPROVED** [review]
Rest LGTM

**Review by BornChanger — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**